Source/DafnyLanguageServer.Test/Various/ShutdownTest.cs

```diff
@@ -0,0 +1,181 @@
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System.Collections.Generic;
+using System.IO.Pipelines;
+using System.Linq;
+using System.Reactive.Concurrency;
+using System.Text;
+using System.Text.Json;
+using System.Threading;
+using Microsoft.CodeAnalysis;
+using OmniSharp.Extensions.JsonRpc;
+using OmniSharp.Extensions.JsonRpc.Client;
+using OmniSharp.Extensions.JsonRpc.Serialization;
+using OmniSharp.Extensions.LanguageServer.Protocol.Client.Capabilities;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+
+namespace Microsoft.Dafny.LanguageServer.IntegrationTest.Various {
+
+  [TestClass]
+  public class ShutdownTest {
+
+    [TestMethod]
+    public async Task LanguageServerStaysAliveIfParentDiesButNoParentIdWasProvided() {
+      var process = await StartLanguageServerRunnerProcess();
+      var error = "";
+      process.ErrorDataReceived += delegate (object sender, DataReceivedEventArgs args) { error += args.Data; };
+
+      var languageServerProcessId = await process.StandardOutput.ReadLineAsync();
```

Comment:
Why do you wrap a custom binary that prints its PID around the server instead of just reading the PID of the regular server?

Reply:
We need to pass the PID of a running process to the language server in the initialize request, and later kill that process. This is what the custom binary is for. It also starts the language server to make the test more realistic, since the PID is supposed to be the parent process of the language server.

Reply:
Then I think your solution makes sense, thanks!